### DISCUSSION TITLE
simulating a simple scenario

### DISCUSSION THREAD

**Discussion by JoseBlanca** [molpopgen/fwdpy11#899] [Q&A] [answered]
Hi:

I'm checking out fwdpy11 because I'm preparing a population genetics introductory course. I think that this simulator could be quite handy, but I'm having problems preparing the most simple simulations.

- Could I initialize a population with two alleles and some desired genotypic frequencies?
- Would it be possible to have just to states for the allele? I mean, could I simulate the mutations as allele a -> allele A and A -> a

Thank your for your work and your time.
Best,

Jose Blanca

**Comment by molpopgen** [comment, marked as answer]
You can initialize a population with two alleles by generating an initial ancestry with msprime to initialize a population.  Then, you may add a (selected) mutation to that population.

Currently, we do not support reverse mutation. We are still modeling infinitely many sites.  Support for more general models is planned, but has no specific timeline.

The primary goal of this software is for research purposes. As such, some of the very simple cases that one may want to use for teaching may actually be a bit harder to do. I actually do not use feature rich simulations for my own teaching. Rather, I tend to use simple things written in JavaScript.

**Reply by molpopgen** [reply]
[This](https://molpopgen.github.io/fwdpy11/pages/types.html?highlight=add_mutation#fwdpy11.DiploidPopulation.add_mutation) is the function for adding mutations to a population with existing ancestry.

**Reply by JoseBlanca** [reply]
Thanks!
